src/cargo/util/config/mod.rs

```diff
@@ -167,6 +167,8 @@ pub struct Config {
     target_dir: Option<Filesystem>,
     /// Environment variables, separated to assist testing.
     env: HashMap<String, String>,
+    /// Environment variables, converted to uppercase to check for case mismatch
+    upper_case_env: HashMap<String, String>,
```

Comment:
The intention of using a HashMap instead of a HashSet was to use the value for the warning output. 

Taking the key.as_env_key() will result in the 'correct' key format.

Reply:
Oh, sorry, I missed that!